From be2d0eaa1e8014c8cbf1268032f09681d3c94b65 Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Mon, 16 Jan 2012 11:01:16 +0800
Subject: [PATCH 191/256] ENGR00170222 vpu: Fix MJPG encoder with rotation
 incorrect issue

MJPEG encode with rotation is incorrect now due to rotMirEnable
bit not set. The patch will fix it.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 vpu/vpu_lib.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/vpu/vpu_lib.c b/vpu/vpu_lib.c
index 327dae3..c72a7e1 100644
--- a/vpu/vpu_lib.c
+++ b/vpu/vpu_lib.c
@@ -1424,6 +1424,8 @@ RetCode vpu_EncStartOneFrame(EncHandle handle, EncParam * param)
 		vpu_mx6q_swreset(0);
 
 	if (pEncInfo->rotationEnable) {
+		if (pCodecInst->codecMode == MJPG_ENC)
+			rotMirEnable = 0x10;    /* Enable rotator */
 		switch (pEncInfo->rotationAngle) {
 		case 0:
 			rotMirMode |= 0x0;
@@ -1443,6 +1445,8 @@ RetCode vpu_EncStartOneFrame(EncHandle handle, EncParam * param)
 		}
 	}
 	if (pEncInfo->mirrorEnable) {
+		if (pCodecInst->codecMode == MJPG_ENC)
+			rotMirEnable = 0x10;    /* Enable mirror */
 		switch (pEncInfo->mirrorDirection) {
 		case MIRDIR_NONE:
 			rotMirMode |= 0x0;
-- 
1.8.0

